external_parser/joiners/example_joiner.cc

```diff
@@ -433,9 +433,12 @@ bool example_joiner::process_joined(v_array<example *> &examples) {
       if (_vw->example_parser->metrics) {
         je->calculate_metrics(_vw->example_parser->metrics.get());
       }
-      if (_binary_to_json &&
-          je->interaction_metadata.payload_type == v2::PayloadType_CB) {
-        log_converter::build_cb_json(_outfile, *je);
+      if (_binary_to_json) {
+        if (je->interaction_metadata.payload_type == v2::PayloadType_CB) {
```

Comment:
could this be:

```
if (_binary_to_json) {
    log_converter::build_json(_outfile, *je);
}
```

and then have `build_json()` check the payload type and call either `build_cb_json` or `build_ccb_json`